src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java

```diff
@@ -174,4 +176,109 @@ private int getFeeds() throws IOException {
     urlConn.disconnect();
     return responseCode;
   }
+
+  @Test
+  public void testHTTPSinkMacroUrl() throws Exception {
+    String inputDatasetName = "input-http-sink-with-macro";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputDatasetName));
+    Map<String, String> properties = new ImmutableMap.Builder<String, String>()
+      .put("url", "${url}")
+      .put("method", "PUT")
+      .put("messageFormat", "Custom")
+      .put("charset", "UTF-8")
+      .put("body", "cask cdap, hydrator tracker, ui cli")
+      .put("batchSize", "1")
+      .put("referenceName", "HTTPSinkReference")
+      .put("delimiterForMessages", "\n")
+      .put("numRetries", "3")
+      .put("followRedirects", "true")
+      .put("disableSSLValidation", "true")
+      .put("connectTimeout", "60000")
+      .put("readTimeout", "60000")
+      .put("failOnNon200Response", "true")
+      .build();
+
+    ImmutableMap<String, String> runtimeProperties =
+      ImmutableMap.of("url", baseURL + "/feeds/users");
+
+    ETLStage sink = new ETLStage("HTTP", new ETLPlugin("HTTP", BatchSink.PLUGIN_TYPE, properties, null));
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    ApplicationManager appManager = deployETL(etlConfig, inputDatasetName);
+
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(inputSchema).set("id", "1").build()
+    );
+    MockSource.writeInput(inputManager, input);
+    // run the pipeline
+    runETLOnce(appManager, runtimeProperties);
+    verifyandcleanupTest();
+  }
+
+  @Test
+  public void testHTTPSinkWithEmptyUrl() {
```

Comment:
If there is exception that is thrown we should rely on junit for that. We should not silently ignore the exception
`@Test(expected=<SomeException>.class)`

Reply:
PR updated accordingly